### PR TITLE
Dsg 2143/separate transaction defintion create and update

### PR DESCRIPTION
**Pull request by afcalderonm**
## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [ ] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

**Review by jcsuarezm — APPROVED** [review]

**Review by Ortega-Dan — APPROVED** [review]
LGTM

**Review by Ortega-Dan — CHANGES_REQUESTED** [review]
Andres, I do not know why this is not failing in the build pipeline, but this doesn't build locally. 
This method was modified some days ago to receive only one argument, but right now 2 are being passed: https://github.com/Nuvalence/dsgov-work-manager/blob/131bdc3aec978845738ca17c69c50ec015a3986c/service/src/main/java/io/nuvalence/workmanager/service/controllers/AdminApiDelegateImpl.java#L407-L409

File: service/src/main/java/io/nuvalence/workmanager/service/mapper/FormConfigurationMapper.java [hunk 1]
```diff
@@ -46,6 +46,7 @@ public interface FormConfigurationMapper {
     @Mapping(
             target = "configuration",
             expression = "java(mapConfigurationAttribute(model.getConfiguration()))")
+    @Mapping(target = "formConfigurationKey", expression = "java(model.getKey())")
```
Comment (Ortega-Dan) [inline service/src/main/java/io/nuvalence/workmanager/service/mapper/FormConfigurationMapper.java:49]: Andres, 
I also realized this custom mapping is not needed if you rather rename the form configuration java variable (even when the table row continues being the same), so the mapping is automatic. It is the same pattern used in transaction definition: https://github.com/Nuvalence/dsgov-work-manager/blob/84af37f0522ba489dda0118e00d80431e920f4f4/service/src/main/java/io/nuvalence/workmanager/service/domain/transaction/TransactionDefinition.java#L55-L56

Comment (afcalderonm) [reply service/src/main/java/io/nuvalence/workmanager/service/mapper/FormConfigurationMapper.java]: Done

**Review by jcsuarezm — APPROVED** [review]

**Pull request merged**